hl7_pims_transformer/tests/test_pid_mapper.py

```diff
@@ -128,3 +128,69 @@ def test_map_pid_29_ts1_datetime_trimming(self) -> None:
                 map_pid(self.original_message, self.new_message)
 
                 self.assertEqual(get_hl7_field_value(self.new_message.pid, "pid_29.ts_1"), expected_value)
+
+    def test_map_pid_3_a28_message_with_n3_prefix(self) -> None:
+        a28_message = (
```

Comment:
Let's try and use the base_message at the top, as well as the parsing and the new message instead of creating them all again in these tests, tests will be more concise and easier to read.

You can override the bits you need like so
 `self.original_message.msh.msh_9.msg_1.value = "ADT"`
 and the PID too with something like: `self.original_message.pid.pid_3[0].value = "1000000001^03^^^NI"
` (adapt to your use case, so it begins with N3 or N4)